packages/main/src/Tree.ts

```diff
@@ -0,0 +1,508 @@
+import UI5Element from "@ui5/webcomponents-base/dist/UI5Element.js";
+import customElement from "@ui5/webcomponents-base/dist/decorators/customElement.js";
+import property from "@ui5/webcomponents-base/dist/decorators/property.js";
+import slot from "@ui5/webcomponents-base/dist/decorators/slot.js";
+import event from "@ui5/webcomponents-base/dist/decorators/event.js";
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { getEffectiveAriaLabelText } from "@ui5/webcomponents-base/dist/util/AriaLabelHelper.js";
+import TreeItem from "./TreeItem.js";
+import TreeItemCustom from "./TreeItemCustom.js";
+import TreeList from "./TreeList.js";
+import ListMode from "./types/ListMode.js";
+import type { ToggleEventDetail as ItemToggleEventDetail, StepInEventDetail, StepOutEventDetail } from "./TreeItemBase.js";
+import type {
+	ClickEventDetail as ListItemClickEventDetail,
+	DeleteEventDetail as ListItemDeleteEventDetail,
+	SelectionChangeEventDetail as ListSelectionChangeEventDetail,
+} from "./List.js";
+
+// Template
+import TreeTemplate from "./generated/templates/TreeTemplate.lit.js";
+
+// Styles
+import TreeCss from "./generated/themes/Tree.css.js";
+import TreeItemBase from "./TreeItemBase.js";
+
+type EventDetail = {
+	item: TreeItemBase,
+}
+type ToggleEventDetail = EventDetail;
+type MouseoverEventDetail = EventDetail;
+type MouseoutEventDetail = EventDetail;
+type ClickEventDetail = EventDetail;
+type DeleteEventDetail = EventDetail;
```

Comment:
Convention for naming these types is ComponentNameEventName... (TreeItemToggleEventDetail). Event types in the list are using the old names..  if possible rename them

Reply:
Thanks a lot for the prompt review! I'll reflect the comments and push an update... 
Just to note regarding the event names of the above comment that I was influenced from the code here: https://github.com/SAP/ui5-webcomponents/blob/main/packages/main/src/List.ts#L1134
but if the preferred convention is ComponentNameEventName, I'll modify accordingly... thnx

Reply:
Done